lib/ansible/plugins/shell/__init__.py

```diff
@@ -141,20 +141,26 @@ def mkdtemp(self, basefile=None, system=False, mode=0o700, tmpdir=None):
 
         return cmd
 
-    def expand_user(self, user_home_path):
+    def expand_user(self, user_home_path, username=''):
         ''' Return a command to expand tildes in a path
 
         It can be either "~" or "~username". We just ignore $HOME
         We use the POSIX definition of a username:
             http://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap03.html#tag_03_426
             http://pubs.opengroup.org/onlinepubs/000095399/basedefs/xbd_chap03.html#tag_03_276
+
+            Falls back to 'current workind directory' as we assume 'home is where the remote user ends up'
         '''
 
         # Check that the user_path to expand is safe
         if user_home_path != '~':
             if not _USER_HOME_PATH_RE.match(user_home_path):
                 # shlex_quote will make the shell return the string verbatim
                 user_home_path = shlex_quote(user_home_path)
+        elif username:
+            # if present the user name is appended to resolve "that user's home"
+            user_home_path += username
+
         return 'echo "%s\t%spwd%s"' % (user_home_path, self._SHELL_SUB_LEFT, self._SHELL_SUB_RIGHT)
```

Comment:
This never worked for me, `echo "~username\tpwd` will return `~username /home/username` as the `~username` is inside double quotes.